src/Common/MetricHelper.cs

```diff
@@ -292,6 +292,30 @@ public void SetPSHost(PSHost host)
         {
         }
 
+        private static void PopulateAuthenticationInfoFromQos(IEnumerable<IAuthenticationInfo> infos, IDictionary<string, string> eventProperties)
```

Comment:
This method is complex. Can you document how it works to help others understand?

Reply:
> This method is complex. Can you document how it works to help others understand?

The method is changed. Please check the new method.